--- /tmp/MetadataParser-170-orig.php	2026-03-11 12:35:41
+++ /tmp/MetadataParser-170-fixed.php	2026-03-11 12:35:52
@@ -151,7 +151,8 @@
                         $classMetadata,
                         $configs,
                         $gqlTypes,
-                        $preProcess
+                        $preProcess,
+                        $container
                     );
                 }
             }
@@ -172,7 +173,8 @@
         Meta $classMetadata,
         array $configs,
         array $gqlTypes,
-        bool $preProcess
+        bool $preProcess,
+        ContainerBuilder $container
     ): array {
         $gqlConfiguration = $gqlType = $gqlName = null;
 
@@ -259,6 +261,9 @@
             case $classMetadata instanceof Metadata\Provider:
                 if ($preProcess) {
                     self::$providers[] = ['reflectionClass' => $reflectionClass, 'metadata' => $classMetadata];
+                    if ($container->hasDefinition($reflectionClass->getName())) {
+                        $container->findDefinition($reflectionClass->getName())->setPublic(true);
+                    }
                 }
 
                 return [];
